mlir-sycl/test/Conversion/SYCLToLLVM/sycl-call-to-llvm.mlir

```diff
@@ -8,6 +8,9 @@
 // CHECK: llvm.func @test() -> [[MEMREF]] {
 // CHECK-NEXT:  %0 = llvm.call @foo() : () -> [[MEMREF]]
 // CHECK-NEXT:  llvm.return %0 : [[MEMREF]]
```

Comment:
The definition of `[[MEMREF]]` probably needs to be changed, as it now should be the promoted type.

Reply:
hmmm....do you know why it is not the promoted type?

Reply:
It is the same type a call is lowered to, give t1.cpp
```
func.func private @foo() -> (memref<?xi32>)

func.func @test() -> (memref<?xi32>) {
  %0 = sycl.call() {Function = @foo, MangledName = @foo, Type = @accessor} : () -> memref<?xi32>
  %1 = call @foo() :  () -> (memref<?xi32>)
  return %0 : memref<?xi32>
}
 ```~/projects/intel-llvm/build  sycl-to-llvm-conv  ./bin/sycl-mlir-opt t1.mlir -convert-sycl-to-llvm                                                                       ✔  etiotto@hds-clx-7 
module {
  llvm.func @foo() -> !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)> attributes {sym_visibility = "private"}
  llvm.func @test() -> !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)> {
    %0 = llvm.call @foo() : () -> !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)>
    %1 = llvm.call @foo() : () -> !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)>
    llvm.return %0 : !llvm.struct<(ptr<i32>, ptr<i32>, i64, array<1 x i64>, array<1 x i64>)>
  }
}

Reply:
I suspect that only arguments are promoted, and not the return type.